ydb/core/kqp/ut/service/kqp_qs_queries_ut.cpp

```diff
@@ -824,7 +824,255 @@ Y_UNIT_TEST_SUITE(KqpQueryService) {
         )", TTxControl::NoTx()).ExtractValueSync();
         UNIT_ASSERT_VALUES_EQUAL_C(result.GetStatus(), EStatus::SUCCESS, result.GetIssues().ToString());
         UNIT_ASSERT(result.GetIssues().Size() == 0);
+    }
+
+    struct ExpectedPermissions {
+        TString Path;
+        THashMap<TString, TVector<TString>> Permissions;
+    };
+
+    Y_UNIT_TEST(DdlPermission) {
+        NKikimrConfig::TAppConfig appConfig;
+        appConfig.MutableTableServiceConfig()->SetEnablePreparedDdl(true);
+        auto setting = NKikimrKqp::TKqpSetting();
+        auto serverSettings = TKikimrSettings()
+            .SetAppConfig(appConfig)
+            .SetKqpSettings({setting});
+
+        TKikimrRunner kikimr(serverSettings);
+        auto db = kikimr.GetQueryClient();
+        auto session = kikimr.GetTableClient().CreateSession().GetValueSync().GetSession();
+
+        const auto checkPermissions = [](NYdb::NTable::TSession& session, TVector<ExpectedPermissions>&& expectedPermissionsValues) {
```

Comment:
We don't change expectedPermissionsValues, so better `const TVector<ExpectedPermissions>& expectedPermissionsValues`